### PR TITLE
feat(ARCO-281): check commit message format and PR title

### PR DESCRIPTION
**Pull request by pawellewandowski98**
## Description of Changes

Added new hook to pre-commit which checks commit message format. 

How to enable it? Run this commands:
- `pip install pre-commit`
- `pre-commit install`
- `pre-commit install --hook-type commit-msg`

Right now during creating commit it checks the format of a commit message.
Requirements: `<type\>(ARCO-XXX): example message` where `<type>` is one of feat, sec, fix, refactor, docs, build, ci, and `(CODE-XXX)` is optional.

image

## Linked Issues / Tickets

ARCO-281

## Testing Procedure

Describe the tests you've added or any testing steps you've taken.

- [ ] I have added new unit tests
- [ ] All tests pass locally
- [ ] I have tested manually in my local environment

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have updated `CHANGELOG.md` with my changes

### CODE REVIEW

File: internal/metamorph/store/README.md
Comment (boecklim) [inline internal/metamorph/store/README.md:1]: I think we can delete this file

Comment (pawellewandowski98) [reply internal/metamorph/store/README.md:1]: Removed

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -6,3 +6,10 @@ repos:
       - id: trailing-whitespace
       - id: end-of-file-fixer
       - id: check-added-large-files
+  - repo: https://github.com/dtaivpp/commit-msg-regex-hook
+    rev: v0.2.2
+    hooks:
+      - id: commit-msg-regex-hook
+        args: [ "--pattern='^(feat|sec|fix|refactor|docs|build|ci)(\\([A-Z]{3,6}-[0-9]{3,6}\\))?: .+$'",
```
[the remark this answers is not in the document]
Comment (pawellewandowski98) [reply .pre-commit-config.yaml]: Added `test`

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -16,29 +16,62 @@ Thank you for considering contributing in the BSV Blockchain ecosystem! This doc
...
+
+`<type>`: Specifies the purpose of the change. Options implemented when sorting changelog:
+```
+deps     -> dependency update
+feat     -> new feature
+sec      -> security update
+fix      -> bug fixes
+refactor -> refactor commits
+docs     -> documentation
+build/ci -> build process update
```
Comment (boecklim) [inline CONTRIBUTING.md:61]: Here is missing
- chore
- test

File: .github/workflows/pr-title-check.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+jobs:
+  check_commit_message:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check commit message format
+        env:
+          PR_TITLE: ${{ github.event.pull_request.title }}
+        run: |
+          if [[ ! "$PR_TITLE" =~ ^(feat|chore|sec|fix|refactor|docs|build|ci|test)(\([A-Z]{3,6}-[0-9]{3,6}\))?:\ .+$ ]]; then
```
Comment (boecklim) [inline .github/workflows/pr-title-check.yml:15]: Here is missing from CONTRIBUTING.md
- deps

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -6,3 +6,10 @@ repos:
       - id: trailing-whitespace
       - id: end-of-file-fixer
       - id: check-added-large-files
+  - repo: https://github.com/dtaivpp/commit-msg-regex-hook
+    rev: v0.2.2
+    hooks:
+      - id: commit-msg-regex-hook
+        args: [ "--pattern='^(feat|chore|sec|fix|refactor|docs|build|ci|test)(\\([A-Z]{3,6}-[0-9]{3,6}\\))?: .+$'",
```
Comment (boecklim) [inline .pre-commit-config.yaml:13]: Here is missing from CONTRIBUTING.md
- deps

File: CONTRIBUTING.md (hunk 2 above)
Comment (pawellewandowski98) [reply CONTRIBUTING.md]: Added

File: .pre-commit-config.yaml (hunk 4 above)
Comment (pawellewandowski98) [reply .pre-commit-config.yaml]: Added

**Review by boecklim — APPROVED** [review]

**Pull request merged**